### PR TITLE
advisor: use json for package database

### PR DESCRIPTION
**Pull request by mvo5**
This PR also simplifies the code a bit by using advisor.Package to be stored in json instead of a helper "snapInfo" struct. 

Based on https://github.com/snapcore/snapd/pull/4845

### CODE REVIEW

File: advisor/backend.go [hunk 1]
```diff
@@ -98,23 +98,37 @@ func Create() (CommandDB, error) {
...
-		if row == nil {
-			row = bname
-		} else {
-			row = append(append(row, ','), bname...)
+		if row != nil {
+			if err := json.Unmarshal(row, &sil); err != nil {
+				return err
+			}
+		}
+		sil = append(sil, Package{Snap: snapName, Version: version})
```
Comment (zyga) [inline advisor/backend.go]: No summary?

Comment (mvo5) [reply advisor/backend.go]: Indeed, I should add a comment. For the mapping of command->snap we do not need the summary, nothing is using that. But that is confusing so a comment is needed here.

**Comment by bboozzoo** [comment]
```
----------------------------------------------------------------------
FAIL: catalogrefresh_test.go:92: catalogRefreshTestSuite.TestCatalogRefresh

catalogrefresh_test.go:112:
    c.Check(dump, DeepEquals, map[string]string{
        "foo": `[{"name":"foo","version":"1.0"}]`,
        "bar": `[{"name":"bar","version":"2.0"}]`,
        "meh": `[{"name":"foo","version":"1.0"},{"name":"bar","version":"2.0"}]`,
    })
... obtained map[string]string = map[string]string{"bar":"[{\"snap\":\"bar\",\"version\":\"2.0\"}]", "foo":"[{\"snap\":\"foo\",\"version\":\"1.0\"}]", "meh":"[{\"snap\":\"foo\",\"version\":\"1.0\"},{\"snap\":\"bar\",\"version\":\"2.0\"}]"}
... expected map[string]string = map[string]string{"foo":"[{\"name\":\"foo\",\"version\":\"1.0\"}]", "bar":"[{\"name\":\"bar\",\"version\":\"2.0\"}]", "meh":"[{\"name\":\"foo\",\"version\":\"1.0\"},{\"name\":\"bar\",\"version\":\"2.0\"}]"}

OOPS: 370 passed, 1 FAILED
```

File: overlord/snapstate/catalogrefresh_test.go [hunk 2]
```diff
@@ -105,10 +105,10 @@ func (s *catalogRefreshTestSuite) TestCatalogRefresh(c *C) {
 	c.Check(osutil.FileExists(dirs.SnapCommandsDB), Equals, true)
 	dump, err := advisor.DumpCommands()
 	c.Assert(err, IsNil)
-	c.Check(dump, DeepEquals, map[string][]string{
-		"foo": {"foo"},
-		"bar": {"bar"},
-		"meh": {"foo", "bar"},
+	c.Check(dump, DeepEquals, map[string]string{
+		"foo": `[{"name":"foo","version":"1.0"}]`,
```
Comment (bboozzoo) [inline overlord/snapstate/catalogrefresh_test.go]: `s/name/snap/` to match the json tags for Package struct

**Comment by zyga** [comment]
Tests are not unit-happy:

```
FAIL: catalogrefresh_test.go:92: catalogRefreshTestSuite.TestCatalogRefresh

catalogrefresh_test.go:108:
    c.Check(dump, DeepEquals, map[string]string{
        "foo": `[{"name":"foo","version":"1.0"}]`,
        "bar": `[{"name":"bar","version":"2.0"}]`,
        "meh": `[{"name":"foo","version":"1.0"},{"name":"bar","version":"2.0"}]`,
    })
... obtained map[string]string = map[string]string{"bar":"[{\"snap\":\"bar\",\"version\":\"2.0\"}]", "foo":"[{\"snap\":\"foo\",\"version\":\"1.0\"}]", "meh":"[{\"snap\":\"foo\",\"version\":\"1.0\"},{\"snap\":\"bar\",\"version\":\"2.0\"}]"}
... expected map[string]string = map[string]string{"foo":"[{\"name\":\"foo\",\"version\":\"1.0\"}]", "bar":"[{\"name\":\"bar\",\"version\":\"2.0\"}]", "meh":"[{\"name\":\"foo\",\"version\":\"1.0\"},{\"name\":\"bar\",\"version\":\"2.0\"}]"}
```

**Pull request merged**